scripts/deploy-tokenizedBallot.ts

```diff
@@ -0,0 +1,45 @@
+import 'dotenv/config';
+import {ethers} from 'hardhat';
+import {TokenizedBallot__factory} from '../typechain-types';
+
+export const deployTokenizedBallot = async (): Promise<void> => {
+  const {ALCHEMY_API_KEY} = process.env;
+  const GOERLI_PRIVATE_KEY = process.env.GOERLI_PRIVATE_KEY as string; // getting linter issues if not assigning type.
+  const [networkName, myTokenAddress, ...proposals] = process.argv.slice(3);
+
+  if (!proposals || proposals.length === 0) {
+    throw new Error('You need to pass a valid constructor argument');
+  }
+  const PROPOSALS_BYTES32 = proposals.map(p => ethers.utils.formatBytes32String(p));
+
+  try {
+    let ContractFactory: TokenizedBallot__factory;
+    let blockNumber: number;
+    if (networkName === 'goerli') {
+      if (!ALCHEMY_API_KEY) {
+        throw new Error('ALCHEMY_API_KEY missing');
+      }
+      const provider = new ethers.providers.AlchemyProvider('goerli', ALCHEMY_API_KEY);
+      const wallet = new ethers.Wallet(GOERLI_PRIVATE_KEY, provider);
+      const signer = wallet.connect(provider);
+      ContractFactory = new TokenizedBallot__factory(signer);
+      const lastBlock = await ethers.provider.getBlock('latest');
+      blockNumber = lastBlock.number - 1; // set this is argument instead? use a blocknumber 12 hours in the future?
+    } else {
+      const accounts = await ethers.getSigners();
+      ContractFactory = new TokenizedBallot__factory(accounts[0]);
+      blockNumber = 0;
+    }
```

Comment:
don't get the point of this `else` block. why do we want to set the block number to zero if the network is not goerli?

Reply:
I did this because we only have one block (0) on hardhat network.
Also, as I understand, we are getting the signer/deployer account in different ways, live network vs local:
```
const provider = new ethers.providers.AlchemyProvider('goerli', ALCHEMY_API_KEY);
const wallet = new ethers.Wallet(GOERLI_PRIVATE_KEY, provider);
const signer = wallet.connect(provider);
```

vs 

`const accounts = await ethers.getSigners();`

Reply:
The problem is you're running this logic as a fallback instead of explicitly checking whether the local network flag has been passed. So if I decided to instead deploy onto ethereum mainnet, i'd be very confused and would have to change this logic.

Reply:
Yeah, that makes sense.
I added `if else` for _mainnet_ (just throws an error for now) and a fallback which also throws an error.